test/Microsoft.TestPlatform.AcceptanceTests/DotnetArchitectureTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if !NETFRAMEWORK
+
+using System.Collections.Generic;
+using System.IO;
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.TestPlatform.AcceptanceTests;
+
+[TestClass]
+public class DotnetArchitectureTests : AcceptanceTestBase
+{
+    public static IEnumerable<object[]> GetRunnerAndDotnetRootEnvVar()
+    {
+        var runnerDataSource = new NetCoreTargetFrameworkDataSource(useDesktopRunner: false);
+        foreach (var entry in runnerDataSource.GetData(null))
+        {
+            yield return new object[] { entry[0], "net6.0", "6.0.100", "DOTNET_ROOT_X86" };
+            yield return new object[] { entry[0], "net6.0", "6.0.100", "DOTNET_ROOT(x86)" };
+            yield return new object[] { entry[0], "net6.0", "6.0.100", "DOTNET_ROOT" };
+
+            yield return new object[] { entry[0], "net5.0", "5.0.100", "DOTNET_ROOT_X86" };
```

Comment:
I am still not 100% sure why this case works locally, I would expect it to fail because that's not understood by . NET 5 and I am correctly pinning SDK version with global.json.